source/common/protobuf/utility.cc

```diff
@@ -505,6 +506,93 @@ std::string MessageUtil::CodeEnumToString(ProtobufUtil::error::Code code) {
   }
 }
 
+namespace {
+
+// Recursive helper method for MessageUtil::redact() below. Note that we have to keep track of
+// whether an ancestor was marked as `sensitive`, not just the field, because of cases like
+// `TlsContext::private_key`, which is of type `core.DataSource` rather than `string`.
+void redactInPlace(Protobuf::Message* message, bool ancestor_is_sensitive) {
+  // If the message is an `Any`, we have to first unpack it to its original type to redact it...
+  auto* any = dynamic_cast<ProtobufWkt::Any*>(message);
+  if (any != nullptr) {
```

Comment:
Do we need to do the same thing for `udpa.types.v1.TypedStrruct`?

Reply:
Done